console/src/components/Services/Settings/utils.ts

```diff
@@ -5,18 +5,41 @@ export const permissionTypes = [
   'delete_permission',
 ];
 
-export const getTableNameFromDef = def => {
-  if (def.constructor.name === 'Object') {
+export type TableDef = {
+  name: string;
+};
+
+export type MetadataObject = {
+  name: string;
+  function_name: string;
+  type: 'table' | 'array_relation' | 'object_relation' | 'function' | 'event_trigger' | 'remote_schema' | 'action';
+  definition: {
+    name: string;
+    role: string;
+    table: {
+      name: string;
+    }
+    configuration: {
+      name: string;
+    }
+  };
+  table_name: string;
+  relationships: Array<{ rel_name: string }>;
+  permissions: Array<{ role_name: string }>
+};
+
+export const getTableNameFromDef = (def: TableDef|string): string => {
```

Comment:
As TypeScript can infer this.
```suggestion
export const getTableNameFromDef = (def: TableDef | string) => {
```